### PR TITLE
Continuously update tree selection in DOM inspection mode

### PR DESCRIPTION
**Pull request by gaearon**
I think this helps explore *what exactly* you're trying to select, as well get a general sense of the hierarchy. To reduce the noise, I throttled the highlighting by 200ms. That also helps avoid highlighting too early when you're just moving the cursor from the panel into the app.

### CODE REVIEW

File: src/backend/agent.js [hunk 1]
```diff
@@ -339,6 +342,7 @@ export default class Agent extends EventEmitter {
     window.addEventListener('mousedown', this._onMouseDown, true);
     window.addEventListener('mouseup', this._onMouseUp, true);
     window.addEventListener('mouseover', this._onMouseOver, true);
+    this._selectInterval = setInterval(this._updateSelectedFiber, 200);
   };
 
   startProfiling = () => {
```
Comment (gaearon) [inline src/backend/agent.js]: This code is in `stopInspectingDOM`. GitHub is messing up the folding.

File: src/backend/agent.js [hunk 2]
```diff
@@ -442,9 +450,20 @@ export default class Agent extends EventEmitter {
     event.stopPropagation();
 
     const target = ((event.target: any): HTMLElement);
-
     // Don't pass the name explicitly.
     // It will be inferred from DOM tag and Fiber owner.
     showOverlay(target);
+
+    const id = this.getIDForNode(target);
+    if (id !== null) {
```
Comment (bvaughn) [inline src/backend/agent.js]: Is the null check correct? If we aren't currently moused over an element that matches a fiber, do we still want to select the most recent one?

File: src/backend/agent.js [hunk 3]
```diff
@@ -359,6 +363,11 @@ export default class Agent extends EventEmitter {
     window.removeEventListener('mousedown', this._onMouseDown, true);
     window.removeEventListener('mouseup', this._onMouseUp, true);
     window.removeEventListener('mouseover', this._onMouseOver, true);
+
+    this._updateSelectedFiber();
```
Comment (bvaughn) [inline src/backend/agent.js]: Is this right? (that we update selection after stop?)

**Comment by bvaughn** [comment]
Btw I think I'm going to write a small throttle+memo util for #74. Maybe we could use it here too, since it seems conceptually similar.

File: src/backend/agent.js (hunk 2 above)
Comment (gaearon) [reply src/backend/agent.js]: Do you think we should clear the selection completely?

File: src/backend/agent.js (hunk 3 above)
Comment (gaearon) [reply src/backend/agent.js]: One last time since the interval is cancelled.

File: src/backend/agent.js (hunk 2 above)
Comment (bvaughn) [reply src/backend/agent.js]: I don't think so. In this case, we should maybe just leave the most recently selected thing as-is?

File: src/backend/agent.js (hunk 3 above)
Comment (bvaughn) [reply src/backend/agent.js]: I guess I was just thinking about the reasons `stopInspectingDOM` might be called:
1. The user clicked on a DOM node (in which case, it's already selected in the tree)
2. The user clicked on the the toggle in the DevTools

Seems like in either case, we wouldn't need to actually call `_updateSelectedFiber` anymore?

Comment (bvaughn) [reply src/backend/agent.js]: Ohh.... I'm wrong about the mouse down bit now I guess.

**Comment by bvaughn** [comment]
Regarding my throttle comment above, maybe it would read a little better to do something like:

```js
import memoize from 'memoize-one';
import throttle from 'lodash.throttle';

class Agent {
  // ...

  _selectFiberForNode = throttle(
    memo((node: HTMLElement) => {
      const id = this.getIDForNode(node);
      if (id !== null) {
        this._bridge.send("selectFiber", id);
      }
    }),
    200
  );
}
```

Then `_onMouseDown` and `_onMouseOver` could both just call `this._selectFiberForNode(target)` directly.

**Comment by gaearon** [comment]
Do you want to do this or I'll do it?

**Comment by bvaughn** [comment]
Don't care, but I think it would be a  nice change to make before merging.

**Comment by bvaughn** [comment]
I'll just do it.

**Pull request merged**